core/dbt/events/functions.py

```diff
@@ -315,7 +310,7 @@ def fire_event(e: Event) -> None:
     global EVENT_HISTORY
     if len(EVENT_HISTORY) == (flags.EVENT_BUFFER_SIZE - 1):
         EVENT_HISTORY.append(e)
-        fire_event(EventBufferFull())
+        EVENT_HISTORY.append(EventBufferFull())
```

Comment:
I don't think this is right. This won't log that the event history is full, it will just append it to the history.

Reply:
Have changed this.